### PR TITLE
CI: Fix downstream anchor projects

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The downstream anchor jobs were disabled because of breaking changes between the monorepo and SPL.

#### Summary of changes

Now that the v2.1 crates are released, and SPL is using them, re-enable the job!

The biggest change is the crate patching, which will hopefully be easier to maintain with this change. The patching functions were essentially copied to mirror the patching scripts in SPL.

### CODE REVIEW

**Comment by joncinque** [comment]
Blocked until #3806 lands

**Review by yihau — APPROVED** [review]

**Pull request merged**
